tests/build_env_test.py

```diff
@@ -399,3 +426,43 @@ def test_run_tests(mocker):
     with pytest.raises(OpenCEError) as exc:
         build_env._run_tests(mock_build_tree, [], conda_env_files)
     assert "There were 4 test failures" in str(exc.value)
+
+def test_build_env_url(mocker):
+    '''
+    This is a negative test of `build_env`, which passes an invalid env file.
+    '''
+    dirTracker = helpers.DirTracker()
+    mocker.patch(
+        'os.mkdir',
+        return_value=0 #Don't worry about making directories.
+    )
+    mocker.patch(
+        'os.system',
+        side_effect=(lambda x: helpers.validate_cli(x, expect=["git clone"], retval=0)) #At this point all system calls are git clones. If that changes this should be updated.
+    )
+    mocker.patch(
+        'os.getcwd',
+        side_effect=dirTracker.mocked_getcwd
+    )
+    mocker.patch(
+        'conda_build.api.render',
+        side_effect=(lambda path, *args, **kwargs: helpers.mock_renderer(os.getcwd(), []))
+    )
+    mocker.patch(
+        'os.chdir',
+        side_effect=dirTracker.validate_chdir
+    )
+    buildTracker = PackageBuildTracker()
+    mocker.patch(
+        'build_feedstock.build_feedstock',
+        side_effect=buildTracker.validate_build_feedstock
+    )
+    mocker.patch(
+        'urllib.request.urlretrieve',
+        side_effect=(lambda x: (os.path.join(test_dir, os.path.basename(x)), None))
+    )
+
+    env_file = 'https://test.com/test-env-invalid1.yaml'
+    with pytest.raises(OpenCEError) as exc:
+        open_ce._main(["build", build_env.COMMAND, env_file])
+    assert "Unexpected key chnnels was found in " in str(exc.value)
```

Comment:
The test is for invalid env file. Is this check correct then?

Reply:
Yeah, this way I can make sure it's using the file I expect it to. I forgot to update the description, so I'll do that.